src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
> Changing IOptions to suddenly start returning different values at different times is definitely unexpected behavior,

Sorry to chime in! If `IOptions` is changed return a value from a `cache`, and the default cache is set up to always return the same value (i.e cache never invalidated) - then how would this be any different to the existing behaviour? Perhaps I am not understanding - if that's the case, feel free to ingore me :-)